test/blackbox-tests/test-cases/github4429.t

```diff
@@ -18,7 +18,6 @@ Create a repository with no HEAD commit:
 At the moment Dune fails, which is bad:
 
   $ dune exec ./main.exe 2>&1 | sed 's/.*\/git/{{ git }}/; s/> .*.output/> {{ output_file }}/g' 
-           git (internal) (exit 128)
-  {{ git }} describe --always --dirty > {{ output_file }}
+           git (internal)
   fatal: bad revision 'HEAD'
```

Comment:
Shouldn't we get rid of this message as well?

Reply:
Yeah, I suppose. I thought it would be nice to warn the user of this issue, but the error message is cryptic anyway.

Reply:
I tried redirecting stderr to `/dev/null` but now I'm getting `# Error: dup2: : Bad file descriptor` everywhere when running `$ dune subst`. No idea why that is.

Reply:
is there a way to test if a git repo has no commit?

Reply:
What I use is `git rev-parse --verify HEAD`. If it succeeds you have a commit, otherwise, no commit.

Reply:
Yes, but it still errors with 128 and returns a cryptic message. So I'm not sure how it helps. It also seems annoying to run an extra command just for this edge case that occurs once a decade.

Reply:
In any case this dup2 error is not right.

Reply:
Can you reproduce this in the test suite with a `(ignore-stderr ..)`?

Reply:
Alright, I just made the redirect using a file.